edb/server/compiler/dbstate.py

```diff
@@ -220,6 +224,16 @@ class QueryUnit:
     cache_key: Optional[uuid.UUID] = None
     cache_sql: Optional[Tuple[bytes, bytes]] = None  # (persist, evict)
 
+    # cache_func_call stores the calling SQL of the cache function
+    # if QueryCacheMode is set to PgFunc.
+    cache_func_call: Optional[Tuple[bytes, bytes]] = None  # (sql, hash)
+
+    # cache_inline is an in-memory temporary reservation of the inlined
+    # query cache when cache_func_call is applied in self.sql and sql_hash,
+    # used by transactions that don't yet have the visibility of the newly
+    # persisted query cache outside the transaction.
+    cache_inline: Optional[Tuple[bytes, bytes]] = None  # (sql, hash)
```

Comment:
Do we ever purge the `cache_inline` data?
One of the goals for the function cache is to avoid the memory usage of caching everything in memory, right?
(Though we might not want to purge it *immediately* after creating the function, because other transactions might want to grab it?)

Reply:
> Do we ever purge the `cache_inline` data?

No, not yet

> ... because other transactions might want to grab it?

Yes, that's my understanding.

I think I can add at least 2 changes to purge this data:

1. Purge in `hydrate_cache()`, this is usually done at server start;
2. Add a new cache registry to store the "inline" cache that rolls forward by some internally-tracked incremental-only transaction ID.

Reply:
We could also prune after some amount of real wall-clock time, which would probably be fine. (Though would sometimes result in having a cache miss in a tx when the inline version is missing)